test/integration/features/profile.feature

```diff
@@ -2,104 +2,161 @@
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
+  Examples: Empty profile name
+    | profilename |
+    | ''          |
+    | ' '         |
+    
+  Scenario Outline: As user, I cannot create profile with special character in profile name
+    When executing "minishift profile set <profilename>" fails
+    Then stderr should contain
     """
-    - minishift	Running		(Active)
+    Profile names must consist of alphanumeric characters only.
     """
-  Scenario: Getting default profile internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
 
-Scenario: Starting Minishift with profile foo
-    When executing "minishift start --profile foo" succeeds
-    Then Minishift should have state "Running"
+  Examples: Wrong profile names
+    | profilename |
+    | '.'         |
+    | '-'         |
+    | '#$'        |
+    | '_VM'       |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
 
-  Scenario: Getting profile foo internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    When executing "minishift profile set <profilename>" succeeds
+    Then profile <profilename> should have state "Does Not Exist"
+     And profile <profilename> should be the active profile
 
-  Scenario: User should be able to list 'foo' and 'minishift'
-      When executing "minishift profile list" succeeds
-      Then stdout should contain
-      """
-      - foo		Running		(Active)
-      - minishift	Running
-      """
+  Examples: Correct profile names
+    | profilename |
+    | Test-123    |
+    | profile_    |
+    | P1_name-    |
+    | vm_45       |
+    | 20          |
+    | random001z  |
+    | 00_XYZ-     |
+    | test_Pro-01 |
+    | foo         |
 
-  Scenario: User should be able set 'minishift' as the active profile
-      When executing "minishift profile set minishift" succeeds
-      Then stdout should contain
-      """
-      Profile 'minishift' set as active profile
-      """
+  Scenario: As user, I can switch between existing profiles
+    When executing "minishift profile set Test-123" succeeds
+    Then profile "Test-123" should be the active profile
+    When executing "minishift profile set minishift" succeeds
+    Then profile "minishift" should be the active profile
+
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is the active profile
```

Comment:
@amitkrout , Still missing this alignment :)

Reply:
@budhrg It should act as a single logic as both the lines are precondition. IMO it should fall under same indentation 
```
Given - precondition
And - precondition
 When - Statement
```
looks more reasonable than 
```
Given - precondition
 And - precondition
 When - statement
```

I am not in the favor of alignment you are suggesting here

Reply:
TBH, I don't really fancy it too much either. However, we do have the same alignment across all feature files. I would open for a discussion to change this, but until then the sensible thing to do is to stick with the alignment we have.

Reply:
FYI, Using the Cucumber feature file formatter in the keywords are left aligned. We have right alignment. I prefer the former, but as said, the latter is what is chosen and should be used until we change it.


This is what we use:
```Cucumber
Scenario: As user, I can start Minishift with default profile 'minishift'
   Given profile "minishift" has state "Does Not Exist"
     And profile "minishift" is the active profile
```

This would be the alternative:
This is what we use:
```Cucumber
Scenario: As user, I can start Minishift with default profile 'minishift'
   Given profile "minishift" has state "Does Not Exist"
   And   profile "minishift" is the active profile
```

Reply:
Influential `The Cucumber Book` uses aligning to left:
```gherkin
Scenario: Successful sign up
New users should get a confirmation email and be greeted
personally by the site once signed in.
  Given I have chosen to sign up
  When I sign up with valid details
  Then I should receive a confirmation email
  And I should see a personalized greeting message
```
We can switch to left. But really this is not a debate for this issue, if somebody wants to switch, we should create a separate issue. In all cases we need to be consistent.

Left or right, but IMHO there is no need to add a meaning to the indentation as the block logic is somehow part of the Gherkin keywords. Also this is the first time I actually realized what's going on with these not aligned steps, before I was just thinking it is chaotic rest from the fast-writing of the feature. Also I do not recall any example using this around Gherkin community.

Reply:
Created #1815 follow-up issue for the formatting.

Reply:
@agajdosi Thanks for creating the issue. 
@budhrg Updated the ```Given And``` indentation suggestion atm. Will update all the feature file once we agree on a common formatting rule cc_ @hferentschik

Reply:
I am against reformatting...